libraries/AP_Math/examples/throttle_curve/throttle_curve.cpp

```diff
@@ -0,0 +1,108 @@
+//
+// Unit tests for the AP_Math throttle curve code
+//
+
+#include <AP_HAL/AP_HAL.h>
+#include <AP_Math/AP_Math.h>
+
+void test_throttle_curve(float alpha);
+void setup();
+void loop();
+
+const AP_HAL::HAL& hal = AP_HAL::get_HAL();
+
+#if CONFIG_HAL_BOARD == HAL_BOARD_SITL
+
+#include <iostream>
+#include <fstream>
+#include <vector>
+using namespace std;
+
+void print_line(std::vector<double> &vec, ofstream &f);
+float copter_throttle_curve(float thr_mid, float throttle_in);
+
+// This code must be copied from Copter's Mode::get_pilot_desired_throttle() since
+// there is no method analogous to AP_Math:throttle_curve in copter
+float copter_throttle_curve(float thr_mid, float throttle_in)
+{
+    const float expo = constrain_float(-(thr_mid-0.5f)/0.375f, -0.5f, 1.0f);
+    // calculate the output throttle using the given expo function
+    float throttle_out = throttle_in*(1.0f-expo) + expo*throttle_in*throttle_in*throttle_in;
+    return throttle_out;
+}
+
+void print_line(vector<double> &vec, ofstream &f)
+{
+    for (double v : vec) {
+        f << v << ",";
+    }
+    f << "\n";
+}
+
+void test_throttle_curve(float alpha)
+{
+    ofstream csv_out;
+
+    const int ns = 101;
+    const int nCurves = 4;
+    float out_mid[nCurves] = {0.2, 0.35, 0.5, 0.7};
+    vector<double> x(ns,0);
```

Comment:
It's a little interesting using double arithmetic here.  That's not what's happening on the flight controller.

It does make for prettier pictures, I guess.